scripts/test_bulk_upsert.py

```
@@ -0,0 +1,441 @@
+            return
+    except KeyError as exc:
+        # bulk_add accesses dict keys directly; malformed items raise KeyError
+        if expect_api_error:
+            print(f"Expected client-side error (KeyError): {exc}")
+            print("---")
+            return
+        print(f"Unexpected KeyError in bulk_add: {exc}")
+        traceback.print_exc()
+        print("---")
+        return
+    except Exception as exc:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)